src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -117,14 +129,27 @@ def compute_reference(frames: np.ndarray,
             frame[:] = shift_frame(frame=frame, dy=dy, dx=dx)
 
         # Create a new reference image from the highest correlated data.
-        ref_image = max_corr_frames.mean(axis=0).astype(np.int16)
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', 'Mean of empty slice')
+            ref_image = np.nanmean(max_corr_frames, axis=0)
         # Shift reference image to position of mean shifts to remove any bulk
         # displacement.
         ref_image = shift_frame(
             frame=ref_image,
             dy=int(np.round(-max_corr_ymax.mean())),
             dx=int(np.round(-max_corr_xmax.mean()))
         )
+        # Clip the reference image back down to the original size and remove
+        # any NaNs remaining. Throw warning if a NaN is found.
+        ref_image = ref_image[pad_y:-pad_y, pad_x:-pad_x]
+        if np.any(np.isnan(ref_image)):
+            print(f"Warning: {np.isnan(ref_image).sum()} NaN value found in "
```

Comment:
If all the frames had the same shift, the lines above this to remove a bulk shift would remove it. The place this could happen is if the frames were shifted in different directions so that the average shift was close to zero, but there was like a corner of the final area that still was "empty".

Reply:
I'll add a quick summary of that.

Reply:
What do you think of this?

`f"Warning: {np.isnan(ref_image).sum()} NaN value found in "
  "reference image after removing padded values during "
  f"iteration {idx}. Likely the image quality is low and "
  " shifting frames failed. Setting NaN values to image mean."`